# Analisa Variable Kategorik

Pada tahapan ini, kita akan menganalisa variable kategorik dengan menjawab pertanyaan-pertanyaan sebagai berikut

Pertama-tama kita import library yang dibutuhkan

In [1]:
import pandas as pd
import numpy as np

Selanjutnya kita import datanya dan kita data dan kolomnya

In [3]:
insurance = pd.read_csv('data/insurance.csv')
insurance

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

**Gender mana yang memiliki tagihan kesehatan paling tinggi?**

In [9]:
tagihan_laki_laki = np.sum(insurance[insurance['sex'] == 'male']['charges'])
tagihan_perempuan = np.sum(insurance[insurance['sex'] == 'female']['charges'])

print(f"Jumlah tagihan laki-laki adalah {tagihan_laki_laki}")
print(f"Jumlah tagihan perempuan adalah {tagihan_perempuan}")

Jumlah tagihan laki-laki adalah 9434763.79614
Jumlah tagihan laki-laki adalah 8321061.194619


Dari data tersebut dapat dilihat bahwa tagihan laki-laki lebih besar daripada perempuan

**Apakah setiap region memiliki jumlah orang yang sama?**

In [14]:
insurance['region'].value_counts()

southeast    364
southwest    325
northwest    325
northeast    324
Name: region, dtype: int64

Dari data di atas, didapatkan bahwasanya jumlah orang tiap region berbeda.

**Mana yang lebih tinggi proporsi perokok atau non perokok?**

In [15]:
insurance['smoker'].value_counts()

no     1064
yes     274
Name: smoker, dtype: int64

Dari data di atas, proporsi data tidak merokok lebih banyak daripada yang merokok.

**Berapa peluang seseorang tersebut adalah perempuan diketahui dia adalah perokok?**

In [24]:
n_perempuan_merokok = len(insurance[(insurance['sex'] == 'female') & (insurance['smoker'] == 'yes')])
n_perempuan_merokok

115

In [25]:
n_perokok = len(insurance[insurance['smoker'] == 'yes'])
n_perokok

274

In [26]:
p_perempuan_given_perokok = n_perempuan_merokok / n_perokok
p_perempuan_given_perokok

0.4197080291970803

Peluang perempuan diketahui dia adalah perokok adalah 0.42

**Berapa peluang seseorang tersebut adalah laki-laki diketahui dia adalah perokok?**

In [27]:
n_laki_merokok = len(insurance[(insurance['sex'] == 'male') & (insurance['smoker'] == 'yes')])
n_laki_merokok

159

In [28]:
n_perokok = len(insurance[insurance['smoker'] == 'yes'])
n_perokok

274

In [29]:
p_laki_given_perokok = n_laki_merokok / n_perokok
p_laki_given_perokok

0.5802919708029197

Peluang perempuan diketahui dia adalah perokok adalah 0.58

**Bagaimana bentuk distribusi peluang besar tagihan dari tiap-tiap region?**